File: sem2_c_cpp_java/Assignment12_27_03_23/q1.java
/*
Java
1. Write a program in which you compare two strings and print if they are same. One String belongs to an abstract class and another belongs to a normal class. Take string inputs from user
*/

import java.util.*;

abstract class QA1{

protected String str1=new String();

public void inputabs(){
	Scanner sc= new Scanner(System.in);
	System.out.println("Enter string for abstract class");
	str1= sc.next();
	
}

};


class Q1 extends QA1{

private String str2=new String();

public Q1(){
	Scanner sc= new Scanner(System.in);
	inputabs();
	System.out.println("Enter string for normal class");
	str2=sc.next();
	if(str2.equals(str1)){
	System.out.println("The strings are same");
	}
	else{
	System.out.println("The strings are different.");
	}
	
}

public static void main(String args[]){
	Q1 b1= new Q1();
}

};

File: sem2_c_cpp_java/Assignment13_04_04_23/q1.cpp
/*
Write a program in which you use operator overloading w.r.t "++" operator to work with an object of a class.
Pre increment operation should search for vowels in a string and post increment operation should match two strings and print
equal or not.
*/
#include<iostream>

using namespace std;

class cls{
private:
string str1, str2;

public:
void operator ++(){
	int i, cntr=0;
	cout<<"Pre increment operator:"<<endl;
	cout<<"Enter a string "<<endl;
	getline(cin,str1);
	cout<<"Vowels present:"<<endl;
	for(i=0; i<str1.length(); i++){
		if(str1[i]=='a' || str1[i]=='e' || str1[i]=='i' || str1[i]=='o'|| str1[i]=='u'||str1[i]=='A' || str1[i]=='E' || str1[i]=='I' || str1[i]=='O'|| str1[i]=='U'){
			cout<< str1[i]<<" "<<endl;
			cntr++;
	}
	}
	if(cntr==0){
		cout<<"No vowels found"<<endl;
	}	
	
	}

void operator ++(int){
	int i, cntr=0;
	cout<<"Post increment operator:"<<endl;
	cout<<"Enter a string "<<endl;
	getline(cin,str1);
	
	cout<<"Enter another string "<<endl;
	getline(cin,str2);
	
	if(str1.compare(str2)==0){
	cout<<"The strings are equal"<<endl;
	}
	else{
	cout<<"The strings are not equal"<<endl;
	}
}


};

int main(){

cls cl;
++cl;
cl++;

return 0;


}
